redbeams/src/main/java/com/sequenceiq/redbeams/service/dbserverconfig/DatabaseServerConfigService.java

```diff
@@ -220,8 +220,18 @@ public DatabaseServerConfig getByCrn(String resourceCrn) {
         return repository.findByResourceCrn(resourceCrn);
     }
 
-    public Optional<DatabaseServerConfig> getByClusterCrn(String clusterCrn) {
-        return repository.findByClusterCrn(clusterCrn);
+    public DatabaseServerConfig getByClusterCrn(String environmentCrn, String clusterCrn) {
+        Optional<DatabaseServerConfig> resourceOpt = getByClusterCrnOptional(environmentCrn, clusterCrn);
+        if (resourceOpt.isEmpty()) {
```

Comment:
you could use `Optional#orElseThrow` to get the value and throw exception

Reply:
nit, but ok